dashboard/src/components/users/EditUserForm/EditUserForm.tsx

```diff
@@ -0,0 +1,472 @@
+      locale: user.locale,
+      defaultRole: user.defaultRole,
+      roles: allAvailableProjectRoles?.map((role) => {
+        const userRole = user.roles.find((sr) => sr.role === role.name);
+        return { role: role.name, value: !!userRole };
+      }),
+    },
+  });
+
+  const {
+    register,
+    formState: { errors, dirtyFields },
+  } = form;
+
+  const isDirty = Object.keys(dirtyFields).length > 0;
+
+  useEffect(() => {
+    onDirtyStateChange(isDirty, 'dialog');
+  }, [isDirty, onDirtyStateChange]);
+
+  useEffect(() => {
+    form.reset(() => ({
+      avatarURL: user.avatarUrl,
+      displayName: user.displayName,
+      email: user.email,
+      emailVerified: user.emailVerified,
+      phoneNumber: user.phoneNumber,
+      phoneNumberVerified: user.phoneNumberVerified,
+      locale: user.locale,
+      defaultRole: user.defaultRole,
+      roles: allAvailableProjectRoles?.map((role) => {
+        const userRole = user.roles.find((sr) => sr.role === role.name);
+        return { role: role.name, value: !!userRole };
+      }),
+    }));
+  }, [user, form, allAvailableProjectRoles]);
+
+  function handleChangeUserPassword() {
+    openDialog('EDIT_USER_PASSWORD', {
+      title: 'Change Password',
+      payload: { user },
+      props: {
+        titleProps: { className: 'mx-auto' },
+        PaperProps: { className: 'max-w-md' },
+      },
+    });
+  }
+
+  const signInMethodsAvailable = user.email || user.userProviders.length !== 0;
+
+  return (
+    <FormProvider {...form}>
+      <Form
+        className="divide-y border-y"
+        onSubmit={(values) => {
+          onEditUser(values, user);
+        }}
+      >
+        <section className="grid grid-flow-col grid-cols-7 p-6">
+          <div className="grid items-center grid-flow-col col-span-6 gap-4 place-content-start">
+            {!user.avatarUrl.includes('default=blank') ? (
+              <Avatar className="w-12 h-12" src={user.avatarUrl} />
+            ) : (
+              <Avatar className="relative inline-flex items-center justify-center w-12 h-12 overflow-hidden bg-gray-300 rounded-full">
+                <span className="text-xs font-medium text-gray-600 uppercase">
+                  {user.displayName.slice(0, 2)}
+                </span>
+              </Avatar>
+            )}
+
+            <div className="grid items-center grid-flow-row">
+              <Text className="text-lg font-medium">{user.displayName}</Text>
+              <Text className="font-normal text-sm+ text-greyscaleGreyDark">
+                {user.email}
+              </Text>
+            </div>
+            {user.disabled && (
+              <Chip
+                component="span"
+                color="error"
+                size="small"
+                label="Banned"
+                className="self-center align-middle"
+              />
+            )}
+          </div>
+          <div>
+            <Select placeholder="Actions" hideEmptyHelperText>
```

Comment:
Let's use a `Dropdown` (using the `menu` prop) here instead of a `Select`. You have plenty of examples to take into consideration.